caas/kubernetes/provider/k8s.go

```diff
@@ -1348,3 +1374,49 @@ func appSecretName(appName, containerName string) string {
 	// A pod may have multiple containers with different images and thus different secrets
 	return "juju-" + appName + "-" + containerName + "-secret"
 }
+
+func mergeConstraintToResources(device devices.KubernetesDeviceParams, resources *core.ResourceRequirements) error {
+	if resources.Limits == nil {
+		resources.Limits = core.ResourceList{}
+	}
+	if resources.Requests == nil {
+		resources.Requests = core.ResourceList{}
+	}
+
+	resourceName := core.ResourceName(device.Type)
+	if v, ok := resources.Limits[resourceName]; ok {
+		logger.Debugf("resource %q - max count %#v has been overwritten by %#v", resourceName, v, device)
```

Comment:
This should never happen right? I think an error may be suitable here as it is not an expected outcome.